tests/utils/test_rest_util.py

```diff
@@ -454,6 +454,34 @@ def test_proxy_conditions_with_no_proxy(self):
             host = "10.0.1.1"
             self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
 
+    def test_proxy_conditions_with_no_proxy_not_set(self):
+        should_use_proxy = True
+        use_proxy = True
+
+        host = "10.0.0.1"
```

Comment:
I think the test should explicitly express that the env variable is not defined (as is, it is actually relying on the current process env)